core/src/main/java/io/cucumber/core/options/CucumberOptionsAnnotationParser.java

```diff
@@ -66,10 +69,14 @@ private void addMonochrome(CucumberOptions options, RuntimeOptionsBuilder args)
         }
     }
 
-    private void addTags(CucumberOptions options, RuntimeOptionsBuilder args) {
+    private void addTags(Class<?> clazz, CucumberOptions options, RuntimeOptionsBuilder args) {
         String tagExpression = options.tags();
         if (!tagExpression.isEmpty()) {
-            args.addTagFilter(tagExpression);
+            try {
+                args.addTagFilter(TagExpressionParser.parse(tagExpression));
+            } catch (TagExpressionException tee) {
+                throw new IllegalArgumentException(String.format("%s at '%s'", tee.getMessage(), clazz.getName()), tee);
```

Comment:
It's generally speaking best to capture exceptions close to the source. So I move the try-catch from the `parse` method here.